using-valtimo/form-flow/create-form-flow-definition.md

```diff
@@ -17,119 +17,157 @@ To create form flow definition, the following steps are necessary:
-of another step defined in this form flow. Since expressions are not yet supported for `nextSteps`, the first step
-will always be selected. So in this example, step 2 will always be followed by step 3.
-
-    ```json
-    {
-        "startStep": "start",
-        "steps": [
-            {
-                "key": "start",
-                "nextStep": "step2"
-            },
-            {
-                "key": "step2",
-                "nextSteps": [
-                    {
-                        "step": "step3"
-                    },
-                    {
-                        "step": "end"
-                    }
-               ]
-            },
-            {
-                "key": "step3",
-                "nextStep": "end"
-            },
-            {
-                "key": "end"
-            }
-        ]
-    }
-    ```
+of another step defined in this form flow.
+
+```json
+{
+    "startStep": "personalDetailsStep",
+    "steps": [
+        {
+            "key": "personalDetailsStep",
+            "nextSteps": [
+                {
+                    "step": "loanApprovedStep"
+                },
+                {
+                    "step": "loanDeniedStep"
+                }
+           ]
+        },
+        {
+            "key": "loanApprovedStep",
+            "nextStep": "summaryStep"
+        },
+        {
+            "key": "loanDeniedStep",
+            "nextStep": "summaryStep"
+        },
+        {
+            "key": "summaryStep"
+        }
+    ]
+}
+```
+
+* The `personalDetailsStep` is now followed by two other steps. This is only allowed when at least one of the two steps
+  is conditional. The order of next step matters. The first `nextStep` with `condition` that is evaluated to `true` will
+  be the next step. When all the conditions are evaluated to `false` the next step will be the default step; which is
+  the step without condition. If no next step is found, the form flow will end. The expression inside the condition is
+  further explained [here](/reference/modules/form-flow.md#step-types).
+
+```json
+{
+    "startStep": "personalDetailsStep",
+    "steps": [
+        {
+            "key": "personalDetailsStep",
+            "nextSteps": [
+                {
+                    "step": "loanApprovedStep",
+                    "condition": "${#jsonPath(step.submissionData, '$.personalDetails.age') >= 21}"
+                },
+                {
+                    "step": "loanDeniedStep"
+                }
+           ]
+        },
+        {
+            "key": "loanApprovedStep",
+            "nextStep": "summaryStep"
+        },
+        {
+            "key": "loanDeniedStep",
+            "nextStep": "summaryStep"
+        },
+        {
+            "key": "summaryStep"
+        }
+    ]
+}
+```
 
 * Configure the step type. Currently, the only step type supported is `form`, which requires a `definition` property to
```

Comment:
Same. If type is required, don't show the example without it. You might get away with adding
```
"type": { ... }
```
in the example above.

Reply:
Type is introduced in this section as to not overwhelm users. I think it's fine to keep it as-is.